qiskit/transpiler/passes/optimization/hoare_opt.py

```diff
@@ -338,4 +348,5 @@ def run(self, dag):
         if self.size > 1:
             for qbt in dag.qubits:
                 self._multigate_opt(dag, qbt.index)
+        self._reset()
```

Comment:
Shouldn't `self._reset` be before `self._initialize`. I'm surprised this works... maybe I'm missing something.

Reply:
You are right, `self._reset` would be better placed before `self._initialize`. Right now it works but we clean the internal state after the run, while we might want to keep the internal state, and clean it only if we call another run.